setup.py

```diff
@@ -30,8 +30,10 @@
 
 EXTRAS_REQUIRE = {
     "neural-compressor": ["neural-compressor>=2.0.0", "onnx", "onnxruntime"],
-    "openvino": ["openvino>=2022.3.0", "transformers>=4.20.0", "onnx"],
-    "nncf": ["nncf"],
+    "openvino": ["openvino-dev>=2022.3.0", "transformers>=4.20.0", "onnx"],
```

Comment:
Why the change from openvino->openvino-dev for the openvino extra?

Reply:
IR pruning requires model optimizer which is bundled in openvino-dev. See [here](https://github.com/vuiseng9/optimum-intel/blob/pr-nncf-jpqd/optimum/intel/openvino/trainer.py#L75).

Reply:
@helena-intel convert_model internally uses serialize, we did attempt to use serialize earlier but it would turn out to be a heavy routine in the OVTrainer to prune the IR and the routine will be a duplicate of convert_model. 

Regarding extra, do you mean we place openvino-dev as part of NNCF keypair? `"nncf": ["nncf", "openvino-dev>=2022.3.0],`

Reply:
I hoped it was just  `apply_pruning_transformation` and other transformations from openvino._offline_transformations and then serialize. Is it more than that? We could see if it is useful to add a generic utils function for that to optimum-intel.

openvino-dev includes quite a few other dependencies which adds some risk of things going wrong and additional unnecessary downloads for users so I prefer to avoid it if it is just for saving a few lines of code. 
If we do need openvino-dev, then yes, I meant it could be part of the "nncf" extra in setup.py, because this is only needed when NNCF is also needed.

Reply:
see [here](https://github.com/vuiseng9/optimum-intel/blob/pr-nncf-jpqd/setup.py#L33-L37)